gtsam/hybrid/tests/testHybridConditional.cpp

```diff
@@ -0,0 +1,73 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/*
+ *  @file testHybridConditional.cpp
+ *  @date Mar 11, 2022
+ *  @author Fan Jiang
+ */
+
+#include <gtsam/hybrid/HybridFactor.h>
+#include <gtsam/hybrid/HybridConditional.h>
+#include <gtsam/hybrid/HybridFactorGraph.h>
+#include <gtsam/hybrid/HybridGaussianFactor.h>
+#include <gtsam/hybrid/HybridBayesNet.h>
+#include <gtsam/hybrid/HybridBayesTree.h>
+
+#include <gtsam/linear/JacobianFactor.h>
+#include <gtsam/inference/BayesNet.h>
+
+#include <CppUnitLite/TestHarness.h>
+
+#include <boost/assign/std/map.hpp>
+using namespace boost::assign;
+
+using namespace std;
+using namespace gtsam;
+
+/* ************************************************************************* */
+TEST_UNSAFE(HybridFactorGraph, test) {
+  HybridConditional test;
+  GTSAM_PRINT(test);
+
+  HybridFactorGraph hfg;
+
+  hfg.add(HybridGaussianFactor(JacobianFactor(0, I_3x3, Z_3x1)));
+  GTSAM_PRINT(hfg);
+}
+
+TEST_UNSAFE(HybridFactorGraph, eliminate) {
+  HybridFactorGraph hfg;
+
+  hfg.add(HybridGaussianFactor(JacobianFactor(0, I_3x3, Z_3x1)));
+
+  auto result = hfg.eliminatePartialSequential({0});
+
+  GTSAM_PRINT(*result.first);
+}
+
+TEST(HybridFactorGraph, eliminateMultifrontal) {
```

Comment:
By employing this scheme we don't need any special code for handling the templated `JunctionTree` and `BayesTree`s, so `eliminateMultifrontal` just work. In contrast, I spent tens of hours to make the old code to compile with `JunctionTree` and failed.